### DISCUSSION TITLE
GDL_Workbench gives unresolved requirement errors

### DISCUSSION THREAD

**Discussion by don-w-p** [gnudatalanguage/gdl#1660] [General]
I ran gdlsetup-Windows-x86_64-standard.exe on a Windows 11 computer and the programs GDL (command Line) and GDL_Workbench were installed.  The command line GDL works but GDL_Workbench gives unresolved requirement errors when starting.  Java, Java SDK, and an EclipseIDE are installed.  The Window features Windows Subsystem for Linux and Virtual Machine Platform are added.  Developer Mode is enabled.  Here is the first part of the log file

!ENTRY org.eclipse.compare 4 0 2023-11-02 10:12:58.554
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: org.eclipse.compare [119]
  Unresolved requirement: Require-Bundle: org.eclipse.ui.forms; bundle-version="[3.2.0,4.0.0)"
    -> Bundle-SymbolicName: org.eclipse.ui.forms; bundle-version="3.11.0.v20201025-0656"; singleton:="true"
       org.eclipse.ui.forms [211]
         Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
         Unresolved requirement: Require-Bundle: org.eclipse.ui; bundle-version="[3.105.0,4.0.0)"; resolution:="optional"
...

**Comment by GillesDuvert** [comment]
Hi @don-w-p I've submitted this as an issue to https://github.com/gnudatalanguage/gdlde .

**Reply by don-w-p** [reply]
Thank you.  See my comment about having found solutions to the problem of unresolved requirement errors when launching.

**Comment by don-w-p** [comment]
Installing the archived jdk-8u202-windows-i586.exe solved this problem and GDL_Workbench now successfully launches,.  When reporting this problem, I only had the latest Java JDK x64 version 21 installed.  Installing the archived Java JDK x64 versions 11 and 17 didn't help.  The archived jdk-8u202-windows-i586.exe can be downloaded from https://www.oracle.com/java/technologies/javase/javase8-archive-downloads.html   There is also an x64 version of this which I didn't try.  Later I discovered that installing OpenJDK20U-jdk_x64_windows_hotspot_20.0.2_9.msi from https://adoptium.net/temurin/archive/?version=20 would also let GDL_Workbench launch.  There is no perspective for debugging, so not everything is fixed.  I speculate that it will only launch with a 32 bit JDK installed.

**Comment by pjb7687** [comment]
Hi, as you can see in this file, GDLDE was compiled with JDK 11:
https://github.com/gnudatalanguage/gdlde/blob/master/.github/workflows/build.yml

I think it is time to update the build with the latest JDK. I will do that right away.

**Reply by don-w-p** [reply]
Thank you for the updated code.  I only looked at your later build with Java 17:   https://github.com/gnudatalanguage/gdlde/releases/tag/v1.0.1   I replied to your next comment.  Please note the conjecture about the gldde-product being win32 and not launching with only an x64 JDK installed but instead requiring a 32 bit JDK be installed..

**Comment by pjb7687** [comment]
This was built with Java 17: https://github.com/gnudatalanguage/gdlde/releases/tag/v1.0.1

**Reply by don-w-p** [reply]
Thank you for the updated code.  The archived jdk-8u202-windows-i586.exe is still needed for the gdlde to launch for the build with JDK 17: https://github.com/gnudatalanguage/gdlde/releases/tag/v1.0.1  The gdlde wouldn't launch for any of the x64 versions 11, 17, and 21 that I tried. Later I discovered that the gdlde would also launch by installing OpenJDK20U-jdk_x64_windows_hotspot_20.0.2_9.msi from https://adoptium.net/temurin/archive/?version=20 There is still no perspective for debugging.  I noticed that the gldde-product itself is win32.  I wonder if it is sometimes looking in the wrong place when a 64 bit JDK is installed.  Java no longer provides win32 builds, only x64 builds for its JDK.    The JDK that it works with is the latest I could find that was 32 bit.
